Tracking/include/Tracking/Reco/TrackExtrapolatorTool.h

```diff
@@ -119,26 +136,33 @@ class TrackExtrapolatorTool {
 
     const auto& ts = first_dis < last_dis ? innermost : outermost;
 
-    // std::cout<<"Selected track state for extrapolation"<<std::endl;
-
     // Get the BoundTrackStateParameters
 
     const auto& surface = ts.referenceSurface();
     const auto& smoothed = ts.smoothed();
+    bool hasSmoothed = ts.hasSmoothed();
+    const auto& filtered = ts.filtered();
     const auto& cov = ts.smoothedCovariance();
 
-    // std::cout<<"Surface::"<<
-    // surface.transform(gctx_).translation()<<std::endl;
-    // std::cout<<"Smoothed::"<<    smoothed.transpose()<<std::endl;
-    // std::cout<<"HasSmoothed::"<< hasSmoothed<<std::endl;
-    // std::cout<<"Filtered::"<<    filtered.transpose()<<std::endl;
-
-    Acts::ActsScalar q = smoothed[Acts::eBoundQOverP] > 0
-                             ? 1 * Acts::UnitConstants::e
-                             : -1 * Acts::UnitConstants::e;
-
-    Acts::BoundTrackParameters sp(surface.getSharedPtr(), smoothed, q, cov);
-
+    if (debug_) {
+      std::cout << "Surface::" << surface.transform(gctx_).translation()
+                << std::endl;
+      if (hasSmoothed)
+        std::cout << "Smoothed::" << smoothed.transpose() << std::endl;
+      std::cout << "HasSmoothed::" << hasSmoothed << std::endl;
+      std::cout << "Filtered::" << filtered.transpose() << std::endl;
+    }
+    Acts::ActsScalar q;
+    if (hasSmoothed)
+      q = smoothed[Acts::eBoundQOverP] > 0 ? 1 * Acts::UnitConstants::e
+                                           : -1 * Acts::UnitConstants::e;
+    else
+      q = filtered[Acts::eBoundQOverP] > 0 ? 1 * Acts::UnitConstants::e
+                                           : -1 * Acts::UnitConstants::e;
```

Comment:
the code that used this is commented out
```suggestion
/*
    Acts::ActsScalar q;
    if (hasSmoothed)
      q = smoothed[Acts::eBoundQOverP] > 0 ? 1 * Acts::UnitConstants::e
                                           : -1 * Acts::UnitConstants::e;
    else
      q = filtered[Acts::eBoundQOverP] > 0 ? 1 * Acts::UnitConstants::e
                                           : -1 * Acts::UnitConstants::e;
*/
```